component/src/main/java/com/celements/struct/classes/TableColumnClass.java

```diff
@@ -0,0 +1,55 @@
+package com.celements.struct.classes;
+
+import java.util.List;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+
+import com.celements.model.classes.AbstractClassDefinition;
+import com.celements.model.classes.fields.ClassField;
+import com.celements.model.classes.fields.StringField;
+import com.celements.model.classes.fields.list.StringListField;
+import com.celements.model.classes.fields.number.IntField;
+
+@Singleton
+@Component(TableColumnClass.CLASS_DEF_HINT)
+public class TableColumnClass extends AbstractClassDefinition implements StructDataClass {
+
+  public static final String SPACE_NAME = "Celements";
+  public static final String DOC_NAME = "StructTableColumnClass";
+  public static final String CLASS_DEF_HINT = SPACE_NAME + "." + DOC_NAME;
+
+  public static final ClassField<String> FIELD_TITLE = new StringField.Builder(CLASS_DEF_HINT,
+      "title").build();
+
+  public static final ClassField<String> FIELD_CONTENT = new StringField.Builder(CLASS_DEF_HINT,
+      "content").build();
+
+  public static final ClassField<Integer> FIELD_ORDER = new IntField.Builder(CLASS_DEF_HINT,
+      "order").build();
+
+  public static final ClassField<List<String>> FIELD_CSS_CLASSES = new StringListField.Builder<>(
+      CLASS_DEF_HINT, "css_classes").multiSelect(true).separator(",").build();
```

Comment:
we want it to be a `ClassField<List<String>>` to make use of it's generic functionality. Otherwise we'd need to duplicate split/join logic from the ClassField into the Table-/ColumnConfig POJOs.

I think the ObjectEdit should support a StringListField where we can write input directly without requiring values from the xclass definition

Reply:
I'am unsure if I understand you correctly. To clarified the AC on the issue.

> "css-class-name" must be free editable values. A static list is not sufficient.

Reply:
I'm not sure if I fully agree with your changed AC. `Celements.StructTableClass.css_classes` should be freely editable yes, but it is intended to represent a **listing** of css class names (a List of Strings). Thus the ClassDefinition Field should reflect this with a correct generic type which is `ClassField<List<String>>`. To solve this we should either:
- add support in `ObjectEdit` to allow free editing of StringListFields if desired (or static values or empty?)
- introduce a new type of StringListField which is backed by `StringClass` instead of `StaticListClass`

Reply:
Like we discussed offline.